tests/test_association_db_sqlite.py

```diff
@@ -151,251 +156,149 @@ def _compare_source_records(self, record_a, record_b):
-                output_src_cat.getSchema().find('id').key)
-
-            for record_a, record_b in zip(
-                    output_src_cat,
-                    input_dia_object.dia_source_catalog):
-                self._compare_source_records(
-                    record_a, record_b)
-
-    def test_get_dia_objects(self):
-        """ Test the retrieval of DIAObjects from the database.
-        """
-        dia_objects = create_test_dia_objects(
-            n_objects=2,
-            n_sources=2,
-            object_centers_degrees=[[0.0, 0.0], [0.1, 0.1]],
-            scatter_arcsec=0.0)
-        dia_collection = DIAObjectCollection(dia_objects)
-        self.assoc_db.store(dia_collection, True)
+        # Get exposure bounding box and load data.
+        bbox = afwGeom.Box2D(self.exposure.getBBox())
+        wcs = self.exposure.getWcs()
+        expMd = pipeBase.Struct(
+            bbox=bbox,
+            wcs=wcs,)
+        output_dia_objects = self.assoc_db.load_dia_objects(expMd)
 
-        self.assoc_db._commit()
+        self.assertEqual(len(output_dia_objects), len(dia_objects))
+        for dia_object, created_object in zip(output_dia_objects, dia_objects):
+            # HTM trixel for this CCD.
+            created_object["indexer_id"] = 225823
+            self._compare_source_records(dia_object, created_object)
 
-        self.assoc_db._db_cursor.execute(
-            "SELECT indexer_id FROM dia_objects")
-        indexer_ids = np.array(
-            self.assoc_db._db_cursor.fetchall(), np.int).flatten()
-
-        output_dia_objects = self.assoc_db._get_dia_objects(indexer_ids)
-
-        for obj_idx in range(2):
-            self.assertEqual(
-                output_dia_objects[obj_idx].n_dia_sources, 2)
-            input_dia_object = dia_collection.get_dia_object(
-                dia_collection.dia_objects[obj_idx].id)
-            self._compare_source_records(
-                output_dia_objects[obj_idx].dia_object_record,
-                input_dia_object.dia_object_record)
-
-            output_src_cat = output_dia_objects[
-                obj_idx].dia_source_catalog
-            output_src_cat.sort(
-                output_src_cat.getSchema().find('id').key)
-
-            for record_a, record_b in zip(
-                    output_src_cat,
-                    input_dia_object.dia_source_catalog):
-                self._compare_source_records(
-                    record_a, record_b)
-
-    def test_get_dia_object_records(self):
-        """ Test the retrieval of SourceRecord objects representing the
-        summarized DIAObjects from the database.
+        # Test overwrite and storage with empty indexer_ids
+        dia_objects = create_test_points(
+            n_points=n_objects,
+            start_id=0,
+            schema=make_minimal_dia_object_schema(),
+            point_locs_deg=object_centers,
+            scatter_arcsec=1.0,
+            indexer_ids=[0 for idx in range(n_objects)],)
```

Comment:
Why not make this a separate test? Or at least loop over test datasets so that you don't have to duplicate code?

Reply:
I've moved more things into their own tests now.